### PR TITLE
john: add gcc11 patch

### PR DESCRIPTION
**Pull request by berbiche**
###### Description of changes

JohnTheRipper cannot build with GCC 11 due to struct/array alignment issues.

See <https://github.com/openwall/john/issues/4604>.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 170013` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>john</li>
  </ul>
</details>

**Comment by applePrincess** [comment]
Can we change homepage and license?
Otherwise change looks good to me.

**Comment by Mindavi** [comment]
Please use fetchpatch on the commit instead: https://github.com/openwall/john/pull/4611/commits/154ee1156d62dd207aff0052b04c61796a1fde3b

**Comment by berbiche** [comment]
Done, switched the patch to use `fetchpatch` and updated the source and homepage url.

File: pkgs/tools/security/john/default.nix [hunk 1]
```diff
@@ -8,11 +8,21 @@ stdenv.mkDerivation rec {
   pname = "john";
   version = "1.9.0-jumbo-1";
 
-  src = fetchurl {
-    url = "http://www.openwall.com/john/k/${pname}-${version}.tar.xz";
-    sha256 = "0fvz3v41hnaiv1ggpxanfykyfjq79cwp9qcqqn63vic357w27lgm";
+  src = fetchFromGitHub {
+    owner = "openwall";
+    repo = pname;
+    rev = "1.9.0-Jumbo-1";
```
Comment (berbiche) [inline pkgs/tools/security/john/default.nix:14]: Should we rename the version string above?
The tag on GitHub is different from the release on the archive.

Comment (Mindavi) [reply pkgs/tools/security/john/default.nix:14]: Either way is fine, I think. What does the Jumbo-1 even mean anyway?

Comment (applePrincess) [reply pkgs/tools/security/john/default.nix:14]: *vernum-jumbo-n* means that this version is community enhanced (e.g. more ciphers can be used) version forked from *vernum* and the current development phase is *n*

Comment (Mindavi) [reply pkgs/tools/security/john/default.nix:14]: Ah like that. Yeah then it definitely sounds like it needs to stay

File: pkgs/tools/security/john/default.nix [hunk 2]
```diff
@@ -70,7 +80,7 @@ stdenv.mkDerivation rec {
   meta = {
     description = "John the Ripper password cracker";
     license = licenses.gpl2;
```
Comment (applePrincess) [inline pkgs/tools/security/john/default.nix:82]: license should be `gpl2Plus`

**Comment by Mindavi** [comment]
Result of `nixpkgs-review pr 170013` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>john</li>
  </ul>
</details>

After the license is updated this is good to go.

**Comment by berbiche** [comment]
Done, thanks.

**Pull request merged**